shared/middleware/files_api.rb

```diff
@@ -51,6 +51,48 @@ def can_view_abusive_assets?(encrypted_channel_id)
     teaches_student?(owner_user_id)
   end
 
+  def report_too_large(quota_type)
+    # Don't record a custom event since these events may be very common.
+    increment_metric('FileTooLarge', quota_type)
+    too_large
+  end
+
+  def quota_half_used?(app_size, body_length)
+    app_size < FilesApi::max_app_size / 2 && app_size + body_length >= FilesApi::max_app_size / 2
```

Comment:
Nit: Based on the code, I think this is only true when we transition from less than half to more than half. However, the method name makes me think it would return true any time we're over half. Not too sure of a better name. Maybe quota_crossed_half_used? or something better.